tools/pot/openvino/tools/pot/algorithms/quantization/accuracy_aware_gna/algorithm.py

```diff
@@ -61,44 +71,126 @@ def _request_alt_statistics(self, model):
+
+                # <= GNA2.0, we changes only fqs for weight, not any fqs for activation
+                is_fq_for_weights = fq_input.type == 'Const'
+                if change_only_weights and not is_fq_for_weights:
+                    logger.debug('Skipping %s because it\'s not weight FQ', fq_name)
                     continue
-                rescale_value = correct_node_overflow(weighted_node, self._oc_stats[add_node_name]['max_per_tensor'])
-                if rescale_value:
-                    logger.debug('Weights and scales for node %s '
-                                 'updated with scale coefficient: %d', weighted_node.name, rescale_value)
-            weight_fq_names.append(fq_name)
-        return model, weight_fq_names, weight_fq_names
+
+                if fq.levels > 2 ** self._default_fq_bit:
+                    logger.debug('Skipping %s because it\'s already in %s precision (levels: %d)',
+                                fq_name, self._precision_change_to, fq.levels)
+                    continue
+
+                # GNA3.5 supports 8/8, 8/16, 16/8, or 16/16 convolution layer
+                if not is_fq_for_weights and fq_output.type != 'Convolution':
+                    continue
+
+                if is_fq_for_weights:
+                    target_nodes.append((None, fq))
+                else:
+                    target_nodes.append((fq, None))
+
+        return target_nodes
+
+    def _convert_fq_to_accuracy_mode(self, model, weights_stats, inputs_stats, fq, is_weights):
+        """ Convert a fq using accuracy config setting
+         :param model: model to apply algo
+         :param weights_stats: stats for weight, only used when is_weights=True
+         :param input_stats: stats for input, only used when is_weights=False
+         :param fq: target fq node to be converted to higher q-bit
+         :param is_weights: True if fq is for weights or False
+         """
+        fq_name = fq.fullname
+        node_quantization_config = {
+            fq_name: self._acc_fq_config[fq_name],
+            'fqs_to_rescale': []
+        }
+        logger.debug('Change bits for %s', fq_name)
+        self._minmax_algo.fill_fq_range(model, weights_stats, inputs_stats, node_quantization_config, self._config)
+        fq.levels = compute_levels(self._acc_fq_config[fq_name], is_weights)
+
+    def _modify_model_in_scope(self, model, nodes_names):
+        self._acc_stats = self._minmax_algo._stats_collector.get_statistics_for_algorithm(self._minmax_algo.name)
+        self._oc_stats = self._stats_collector.get_statistics_for_algorithm(self._oc_algo.name)
+        
+        fqs_to_modify = self._get_fq_to_modify(model, nodes_names)
+
+        converted_fq_names = []
+        for fq_for_activation, fq_for_weight in fqs_to_modify:
+            if fq_for_activation is not None:
+                acc_stats = self._minmax_algo._stats_collector.get_statistics_for_algorithm(self._minmax_algo.name)
```

Comment:
Are these statistics the same as the statistics collected in line 159? If yes, can we somehow improve its usage?

Reply:
Yes, they are the same as you said. so I modified it to call 'get_statistics_for_algorithm' only at the beginning.